File: src/ruby/lib/dependabot/job/models/enums/update_type.rb
# typed: strict
# frozen_string_literal: true

require 'sorbet-runtime'

module Dependabot
  module Job
    module Models
      module Enums
        # Represents the type of update allowed for dependencies.
        class UpdateType < T::Enum
          enums do
            # Security updates only.
            Security = new('security')

            # Version updates including patch, minor, and major versions.
            Version = new('version')

            # All types of updates.
            All = new('all')
          end
        end
      end
    end
  end
end

File: src/dotnet/src/Dependabot.Job/Models/Enums/UpdateType.cs
using System.Text.Json.Serialization;

namespace Dependabot.Job.Enums;

/// <summary>
/// Represents the type of update allowed for dependencies.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter<UpdateType>))]
public enum UpdateType
{
  /// <summary>
  /// Security updates only.
  /// </summary>
  Security,

  /// <summary>
  /// Version updates including patch, minor, and major versions.
  /// </summary>
  Version,

  /// <summary>
  /// All types of updates.
  /// </summary>
  All
}
